plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -3012,9 +3024,16 @@ protected void saveTransOrJob( ISharedObjectsTransformer objectTransformer, Repo
               RepositoryFile.DEFAULT_LOCALE, element.getName() ).createdDate(
               versionDate != null ? versionDate.getTime() : new Date() ).description( RepositoryFile.DEFAULT_LOCALE,
               Const.NVL( element.getDescription(), "" ) ).build();
-      file =
+      try {
+        file =
           pur.createFile( element.getRepositoryDirectory().getObjectId().getId(), file,
-              new NodeRepositoryFileData( objectTransformer.elementToDataNode( element ) ), versionComment );
+            new NodeRepositoryFileData( objectTransformer.elementToDataNode( element ) ), versionComment );
+      } catch ( SOAPFaultException e ) {
+        if ( e.getMessage().contains( UnifiedRepositoryCreateFileException.PREFIX ) ) {
+          throw new KettleException(
+            "File " + file.getName() + " was already created by another process" );
```

Comment:
Shouldn't this be in the message bundle instead of hard-coded and non-NLS-izable?